src/utils/string_helpers.py

```diff
@@ -1,17 +1,34 @@
 from typing import Tuple
 
 
-def split_absolute_ref(reference: str) -> Tuple[str, str, str]:
-    """Split a reference into data_source, path and attribute.
+def split_address(address: str) -> Tuple[str | None, str, str, str]:
```

Comment:
Perhaps a bit inconsistent that protocol is None if the address contains no protocol, but the path and attribute are empty strings